### PR TITLE
NINOA-87

### PR DESCRIPTION
**Pull request by DurgaJonnalagadda**

### CODE REVIEW

File: app/v1/models/request/Postcode.scala [hunk 1]
```diff
@@ -17,27 +17,31 @@
 package v1.models.request
 
 import play.api.Logger
-import play.api.libs.json.{JsString, Reads, Writes, __}
-
-import scala.util.matching.Regex
+import play.api.libs.json.{JsPath, JsString, JsonValidationError, Reads, Writes, __}
```
Comment (TomDudley92) [inline app/v1/models/request/Postcode.scala]: Can you remove `JsPath` from the imports please

File: app/v1/models/request/AddressModel.scala [hunk 2]
```diff
@@ -44,52 +44,34 @@ object AddressModel {
...
-    countryCode.fold(postcode)(
-      countryCde => countryCde.toUpperCase match {
-        case "GBR" => postcode.fold({
-          Logger.warn("[AddressModel][checkPostcodeMandated] - postcode is required if country code is GBR")
-          throw new IllegalArgumentException("Postcode required if Country code is GBR")
-        })(postcode => Some(postcode))
-        case _ => postcode
-      }
-    )
+  private[models] def postcodeValidation(postcode: Option[Postcode], countryCode: Option[String]): Boolean = {
```
Comment (TomDudley92) [inline app/v1/models/request/AddressModel.scala]: This file needs to be reverted to what is in master as this code had leaked from another branch

**Comment by TomDudley92** [comment]
Sorry, this is not approved. See required changes

File: app/v1/models/request/AddressModel.scala (hunk 2 above)
Comment (DurgaJonnalagadda) [reply app/v1/models/request/AddressModel.scala]: reverted to what is in the master

File: app/v1/models/request/Postcode.scala (hunk 1 above)
Comment (DurgaJonnalagadda) [reply app/v1/models/request/Postcode.scala]: done

**Pull request merged**